cypress/integration/core_opensearch_dashboards/opensearch_dashboards/apps/query_enhancements/histogram_interaction.spec.js

```diff
@@ -0,0 +1,223 @@
+import { NEW_SEARCH_BUTTON } from '../../../../../utils/dashboards/data_explorer/elements.js';
+import {
+  generateAllTestConfigurations,
+  getRandomizedWorkspaceName,
+  setDatePickerDatesAndSearchIfRelevant,
+} from '../../../../../utils/apps/query_enhancements/shared';
+import { generateHistogramInteractionTestConfiguration } from '../../../../../utils/apps/query_enhancements/histogram_interaction';
+const workspace = getRandomizedWorkspaceName();
+
+describe('histogram interaction', { testIsolation: true }, () => {
+  beforeEach(() => {
+    // Load test data
+    cy.setupTestData(
+      SECONDARY_ENGINE.url,
+      ['cypress/fixtures/query_enhancements/data_logs_1/data_logs_small_time_1.mapping.json'],
+      ['cypress/fixtures/query_enhancements/data_logs_1/data_logs_small_time_1.data.ndjson']
+    );
+
+    // Add data source
+    cy.addDataSource({
+      name: DATASOURCE_NAME,
+      url: `${SECONDARY_ENGINE.url}`,
+      authType: 'no_auth',
+    });
+    // Create workspace
+    cy.deleteWorkspaceByName(workspace);
+    cy.visit('/app/home');
+    cy.osd.createInitialWorkspaceWithDataSource(DATASOURCE_NAME, workspace);
+    cy.wait(2000);
+    cy.createWorkspaceIndexPatterns({
+      workspaceName: workspace,
+      indexPattern: INDEX_PATTERN_WITH_TIME.replace('*', ''),
+      timefieldName: 'timestamp',
+      indexPatternHasTimefield: true,
+      dataSource: DATASOURCE_NAME,
+      isEnhancement: true,
+    });
+    cy.navigateToWorkSpaceSpecificPage({
+      url: BASE_PATH,
+      workspaceName: workspace,
+      page: 'discover',
+      isEnhancement: true,
+    });
+    cy.getElementByTestId(NEW_SEARCH_BUTTON).click();
+  });
+
+  afterEach(() => {
+    cy.deleteWorkspaceByName(workspace);
+    cy.deleteDataSourceByName(DATASOURCE_NAME);
+    // TODO: Modify deleteIndex to handle an array of index and remove hard code
+    cy.deleteIndex(INDEX_WITH_TIME_1);
+  });
+
+  generateAllTestConfigurations(generateHistogramInteractionTestConfiguration).forEach((config) => {
+    const selectLangs = () => {
+      let langs = [];
+      if (config.datasetType === 'INDEXES') {
+        langs = ['PPL'];
+      } else {
+        langs = ['DQL', 'Lucene', 'PPL'];
+        langs.splice(langs.indexOf(config.language), 1); // remove current lang to iterate over the other two only
+      }
+      return langs;
+    };
+
+    it(`check histogram visibility for ${config.testName}`, () => {
+      cy.setDataset(config.dataset, DATASOURCE_NAME, config.datasetType);
+      cy.setQueryLanguage(config.language);
+      setDatePickerDatesAndSearchIfRelevant(config.language);
+      const assertionByLang = {
+        DQL: 'be.visible',
+        Lucene: 'be.visible',
+        'OpenSearch SQL': 'not.exist',
+        PPL: 'be.visible',
+      };
+      const assertion = assertionByLang[config.language];
+      cy.getElementByTestId('dscChartChartheader').should(assertion);
+      cy.getElementByTestId('discoverChart').should(assertion);
```

Comment:
I would suggest refactoring lines 81-89 to:

```
      // check the visibility of histogram
      if (config.histogram) {
        cy.getElementByTestId('dscChartChartheader').should('be.visible');
        cy.getElementByTestId('discoverChart').should('be.visible');
      } else {
        cy.getElementByTestId('dscChartChartheader').should('not.exist');
        cy.getElementByTestId('discoverChart').should('not.exist');
      }
 ```

This way if the support for histogram changes in the future, we only need to change it in one area